src/System.Runtime.Numerics/tests/BigInteger/PerformanceTests.cs

```diff
@@ -0,0 +1,198 @@
+        [InlineData(1000000, 64, 16)]
+        [InlineData(1000000, 256, 128)]
+        [InlineData(100000, 1024, 512)]
+        [InlineData(10000, 4096, 2048)]
+        [InlineData(1000, 16384, 8192)]
+        [InlineData(100, 65536, 32768)]
+        public void Remainder(int count, int leftBits, int rightBits)
+        {
+            RunBenchmark(count, leftBits, rightBits, (l, r) => BigInteger.Remainder(l, r));
+        }
+
+        [Theory]
+        [ActiveIssue("PerformanceTest")]
+        [InlineData(100000, 16, 16, 16)]
+        [InlineData(10000, 64, 64, 64)]
+        [InlineData(1000, 256, 256, 256)]
+        [InlineData(100, 1024, 1024, 1024)]
+        [InlineData(10, 4096, 4096, 4096)]
+        [InlineData(1, 16384, 16384, 16384)]
+        public void ModPow(int count, int leftBits, int rightBits, int otherBits)
+        {
+            RunBenchmark(count, leftBits, rightBits, otherBits, (l, r, o) => BigInteger.ModPow(l, r, o));
+        }
+
+        private void RunBenchmark(int count, int bits, Action<BigInteger> operation)
+        {
+            BigInteger[] value = CreateIntegerSeed(count, bits);
+
+            long result = RunBenchmark(count, i => operation(value[i]));
+
+            _output.WriteLine("({1:N0}) : {2:N0} ms / {0:N0} ops", count, bits, result);
+        }
+
+        private void RunBenchmark(int count, int leftBits, int rightBits, Action<BigInteger, BigInteger> operation)
+        {
+            BigInteger[] left = CreateIntegerSeed(count, leftBits);
+            BigInteger[] right = CreateIntegerSeed(count, rightBits);
+
+            long result = RunBenchmark(count, i => operation(left[i], right[i]));
+
+            _output.WriteLine("({1:N0}, {2:N0}) : {3:N0} ms / {0:N0} ops", count, leftBits, rightBits, result);
+        }
+
+        private void RunBenchmark(int count, int leftBits, int rightBits, int otherBits, Action<BigInteger, BigInteger, BigInteger> operation)
+        {
+            BigInteger[] left = CreateIntegerSeed(count, leftBits);
+            BigInteger[] right = CreateIntegerSeed(count, rightBits);
+            BigInteger[] other = CreateIntegerSeed(count, otherBits);
+
+            long result = RunBenchmark(count, i => operation(left[i], right[i], other[i]));
+
+            _output.WriteLine("({1:N0}, {2:N0}, {3:N0}) : {4:N0} ms / {0:N0} ops", count, leftBits, rightBits, otherBits, result);
+        }
+
+        private const int MAX_SEED = 10;
+        private const int RUN_COUNT = 3;
+
+        private long RunBenchmark(int count, Action<int> operation)
+        {
+            operation(0);
+
+            long result = long.MaxValue;
+            for (int j = 0; j < RUN_COUNT; j++)
+            {
+                Stopwatch watch = new Stopwatch();
+
+                watch.Start();
+                for (int i = 0; i < count; i++)
+                    operation(i % MAX_SEED);
+                watch.Stop();
+
+                result = Math.Min(result, watch.ElapsedMilliseconds);
```

Comment:
The idea is, that one run isn't deterministic enough, since there could be some kind of noise. So each benchmark gets three opportunities to prove its point.